lib/tmdb/client.rb

```diff
@@ -254,36 +289,43 @@ def get_tv_episode_data(series_id:, season_number:, episode_number:)
       def request(endpoint, params)
         return { results: [] } if params[:query].present? && searchable_query(params[:query]).empty?
 
-        api_path = case endpoint
-          when :credits_data then "/credit/#{params[:credit_id]}?api_key=#{API_KEY}"
-          when :person_data then "/person/#{params[:person_id]}?api_key=#{API_KEY}"
-          when :person_search then "/search/person?api_key=#{API_KEY}&query=#{searchable_query(params[:query])}"
-          when :person_movie_credits then "/person/#{params[:person_id]}/movie_credits?api_key=#{API_KEY}"
-          when :person_tv_credits then "/person/#{params[:person_id]}/tv_credits?api_key=#{API_KEY}"
-          when :movie_search then "/search/movie?api_key=#{API_KEY}&query=#{searchable_query(params[:query])}"
-          when :movie_data then "/movie/#{params[:movie_id]}?api_key=#{API_KEY}&append_to_response=trailers,credits,releases"
-          when :tv_series_search then "/search/tv?api_key=#{API_KEY}&query=#{searchable_query(params[:query])}"
-          when :tv_series_data then "/tv/#{params[:series_id]}?api_key=#{API_KEY}&append_to_response=credits"
-          when :tv_season_data then "/tv/#{params[:series_id]}/season/#{params[:season_number]}?api_key=#{API_KEY}&append_to_response=credits"
-          when :tv_episode_data then "/tv/#{params[:series_id]}/season/#{params[:season_number]}/episode/#{params[:episode_number]}?api_key=#{API_KEY}"
-          when :multi_search then "/search/multi?api_key=#{API_KEY}&query=#{searchable_query(params[:query])}"
-          when :discover_search then url_for_movie_discover_search(params)
+        api_path = build_url(endpoint, params)
+        response = open("#{BASE_URL}#{api_path}").read
+        JSON.parse(response, symbolize_names: true)
+      end
+
+      def build_url(endpoint, params)
```

Comment:
I pulled this out into a separate method partially for ease of testing and partially to make the `request` method concisely about making the request and parsing the JSON. Just a quick out-and-back.

Reply:
👍 Yeah, I like that approach